### DISCUSSION TITLE
[Docker] Why 404 NotFound error in Ocelot gateway?

### DISCUSSION THREAD

**Discussion by mukeshsjpl** [ThreeMammals/Ocelot#1829] [Q&A] [answered]
hello,
 I am facing problem in route,

error is 404 not found where my DownstreamPathTemplate is working fine

my routes is :
```json
{
      "UpstreamPathTemplate": "/hr/employee/{id}/family",
      "UpstreamHttpMethod": [ "GET" ],
      "DownstreamPathTemplate": "/api/employee/family/{id}/getAllByEmployeeId",
      "DownstreamScheme": "http",
      "RouteIsCaseSensitive": false,
      "DownstreamHostAndPorts": [
        {
          "Host": "43.251.72.44",
          "Port": 8092
        }
      ]
    },
```
 screenshot attached
![error](https://github.com/ThreeMammals/Ocelot/assets/143499663/f3fbf286-b82d-4a8a-9e8c-bce8cdb610e7)
![working](https://github.com/ThreeMammals/Ocelot/assets/143499663/44281b81-d115-464b-9b36-fbba210bf07b)

I am not able to find the actual problem...

please help me.

thanks

**Comment by raman-m** [comment, marked as answer]
Attach your log for this 404 problem!!! It is hard to say based on route config only 😄  I see that direct connections are OK.
It could be a Docker problem if you use Docker environment... You use the same host but ports are different.
It could be Docker network DNS problem...
Do you use Docker to host gateway and downstream service apps?

Also, [your repos](https://github.com/mukeshsjpl?tab=repositories) are empty! Usually we don't serve ghosts!
First, fork Ocelot repo and start contributing!

**Reply by mukeshsjpl** [reply]
Thank you for your valuable support, There was a route definition problem, now its solve.

**Reply by raman-m** [reply]
You are welcome!
Could you start contribution to Ocelot?
But first, fork Ocelot repository into your account plz.
